client/src/components/ItemDetail/ItemDetail.tsx

```diff
@@ -21,11 +27,25 @@ function ItemDetail({
 }) {
   const [curAmount, setCurAmount] = useState<number>(0);
   const [isRemoving, setIsRemoving] = useState<boolean>(false);
+  const [isOverflow, setIsOverflow] = useState<boolean>(false);
+  const [mediaCollapsed, setMediaCollapsed] = useState<boolean>(false);
+  const [minHeightMarginTop, setMinHeightMarginTop] = useState<number>(0);
 
   const theme = useTheme();
 
   const item = cartItem ? cartItem.item : emptyItem();
-  const cornerRoundingPercentage = 10;
+  const cornerRoundingRadius = "24px";
+
+  // Clamp minimum height for footer vh
+  const footerMinHeight = 180;
+  let footerPercentageHeight = 0.4;
+  if (
+    document.documentElement.clientHeight * footerPercentageHeight <
+    footerMinHeight
+  ) {
+    footerPercentageHeight =
+      footerMinHeight / document.documentElement.clientHeight;
+  }
```

Comment:
Have you tried some css options instead of doing this programatically?

Reply:
The fix here is for smooth animation for the expand/collapse. Prior tried approach to setting minHeight for the footer contents results in margin snapping back to 0 on start of `transition` animation (margin needed to be negative to offset the increased div height - this was where the obsolete `minHeightMarginTop` came from).